config/confighttp/confighttp.go

```diff
@@ -132,6 +136,15 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 		transport.IdleConnTimeout = *hcs.IdleConnTimeout
 	}
 
+	// Setting the Proxy URL
+	if hcs.ProxyURL != "" {
+		proxyURL, err := url.Parse(hcs.ProxyURL)
```

Comment:
Shouldn't this be rather [url.ParseRequestURI](https://pkg.go.dev/net/url#ParseRequestURI)?
```suggestion
		proxyURL, err := url.ParseRequestURI(hcs.ProxyURL)
```